lib/charms/tempo_k8s/v1/charm_tracing.py

```diff
@@ -127,11 +132,29 @@ def get_current_span() -> Union[Span, None]:
     return cast(Span, span)
 
 
+def _get_tracer_from_context(ctx: Context) -> Optional[ContextVar]:
+    tracers = [v for v in ctx if v is not None and v.name == "tracer"]
+    if tracers:
+        return tracers[0]
+    return None
+
+
 def _get_tracer() -> Optional[Tracer]:
+    """Find tracer in context variable and as a fallback locate it in the full context."""
     try:
         return tracer.get()
     except LookupError:
-        return None
+        try:
+            logger.debug("tracer was not found in context variable, looking up in context")
```

Comment:
Not sure, but should it be:
```suggestion
            logger.debug("tracer was not found in context variable, looking up in default context")
```

Reply:
Sounds better, updated.